controllers/secret_controller.go

```diff
@@ -42,7 +42,8 @@ const (
 type SecretSyncController struct {
 	ClientCache *clientcache.ClientCache
 	// TODO We need a better structure for empty selectors (match whole kind)
-	selectors map[types.NamespacedName]labels.Selector
+	selectorMutex sync.Mutex
```

Comment:
A `sync.Mutex` must never be copied, and as such, should preferably be declared as a pointer `*sync.Mutex`.

Reply:
I disagree with this, ``sync.Mutex`` without the pointer is the format as recommended in Effective Go.

Reply:
>sync.Mutex without the pointer is the format as recommended in Effective Go

Can you share a reference? And not just https://golang.org/doc/effective_go 😉

Reply:
https://golang.org/doc/effective_go#allocation_new ;)

Basically, it's because sync.Mutex valid state is zero-value, and thus everytime you create a struct of a pointer to struct, it works straight away: "The zero-value-is-useful property works transitively.". 

If it was *sync.Mutex, it wouldn't. As far as I can find, all instances of sync.Mutex in a struct are without pointer in Golang's source code: https://github.com/golang/go